### PR TITLE
RELENG-167: Add support for signing system extensions in iscript

### PR DESCRIPTION
**Pull request by bhearsum**
This has been tested for MozillaVPN, but not for other products. It's hard to imagine how this could break Firefox though.

### CODE REVIEW

File: iscript/src/iscript/mac.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def get_app_dir(parent_dir):
         UnknownAppDir: if there is no single app dir
 
     """
-    apps = glob("{}/*.app*".format(parent_dir)) + glob("{}/*/*.app*".format(parent_dir))
+    apps = glob("{}/*.app*".format(parent_dir)) + glob("{}/*/*.app*".format(parent_dir)) + glob("{}/*.systemextension*".format(parent_dir))
```
Comment (escapewindow) [inline iscript/src/iscript/mac.py:380]: I'm ok with this. I could also see `iscript.constants.MAC_PRODUCT_CONFIG.PRODUCT.app_globs` or `app_extensions` if we need to customize this anymore :)

File: iscript/src/iscript/mac.py [hunk 2]
```diff
@@ -926,7 +926,7 @@ async def create_pkg_files(config, sign_config, all_paths):
     for app in all_paths:
         # call set_app_path_and_name because we may not have called sign_app() earlier
         set_app_path_and_name(app)
-        app.pkg_path = app.app_path.replace(".appex", ".pkg").replace(".app", ".pkg")
+        app.pkg_path = app.app_path.replace(".appex", ".pkg").replace(".app", ".pkg").replace(".systemextension", ".pkg")
```
Comment (escapewindow) [inline iscript/src/iscript/mac.py]: Interesting, we want to create .pkg files for the system extension(s)?
If not, we could also skip `.systemextension` apps during pkg creation.

**Review by escapewindow — APPROVED** [review]
I suppose we'll see if someone adds a `.systemextension` to Firefox?
I don't see one today. Otherwise, I might have asked you to do a manual run of an l10n `mac_notarize` task.

File: iscript/src/iscript/mac.py (hunk 2 above)
Comment (bhearsum) [reply iscript/src/iscript/mac.py]: We don't really want to (the very first thing we do with these is unpacking them to get the actual bits out), but this is what iscript spits out right now. Ideally (at least, IMO) we'd spit out the same format as the input.

**Comment by bhearsum** [comment]
> I suppose we'll see if someone adds a .systemextension to Firefox?

If we do that I _think_ we'll need to start signing that separately from Firefox (like we do for the MozillaVPN components right now). I could be wrong though.

File: iscript/src/iscript/mac.py (hunk 2 above)
Comment (escapewindow) [reply iscript/src/iscript/mac.py]: iscript spits out a tarball and a pkg. if we only extract it from the tarball, we can probably skip the pkg ?

Comment (bhearsum) [reply iscript/src/iscript/mac.py]: Oh god, I can't believe I missed that (pkgs are listed last in the log output, which is the only thing I saw). Yes, let's skip pkg generation for mozillavpn! I'll need to update the braindump scripts + this patch to do that.

Comment (escapewindow) [reply iscript/src/iscript/mac.py]: Ok. Non-blocker. We waste a tiny bit of compute time and a tiny bit of disk space/network bandwidth for the pkg, but if we end up landing as-is I'm ok with it.

Comment (bhearsum) [reply iscript/src/iscript/mac.py]: Too late, I already did this! I'll have a new version up for review shortly.

**Comment by bhearsum** [comment]
I tested this with a recent notarize_1 and notarize_3 jobs from mozilla-beta, as well as with both distribution and developer id mozillavpn signing. They all worked fine.

**Comment by escapewindow** [comment]
LGTM! Thank you!

**Pull request merged**